test6.go

```diff
@@ -1,61 +1,56 @@
 package main
 
 import (
-    "fmt"
-    "io/ioutil"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
 	"net/http"
 	"os"
-	"log"
-	"io"
-	"encoding/json"
-	"encoding/csv"
 )
 
-type Sheet struct{
-	Sitename string `json:"sitename"`
-	Location string `json:"location"`
-	Address *Address `json:address`
-
+type Sheet struct {
+	Sitename string   `json:"sitename"`
+	Location string   `json:"location"`
+	Address  *Address `json:address`
 }
 
 type Address struct {
 	Number string `json:"number"`
 }
 
-
-
 func main() {
 
 	csvFile, _ := os.Open("test.csv")
```

Comment:
Welcome to Go, the programming language where you should *never* ignore errors (unless there is a very good reason to)
In this example, what happens if `test.csv` doesn't exists?

You should do something like this:

```go
f, err := os.Open("test.csv")
if err != nil {
  fmt.Printf("Error while opening test.csv: %v\n", err)
  return
}
```

Or since we are one the main function and this is the beginning of the program, we can just crash

```go
f, err := os.Open("test.csv")
if err != nil {
  log.Fatal(err)
}
```

Reply:
I initially had an error message but removed it because it seemed like clutter, as there is no way test.csv couldn't exist on my computer, but I'll be sure to add more error catching as it is good practice

Reply:
Actually it's not only good practice, it's extremely important:
- even on your own computer, you can compile your program using `go install` and run it from a directory where there is no `test.csv` file
- the rest of the code keeps going even if `f` is `nil`, which at best will make your program crash with an error that has nothing to do with the original one, making it hard to understand, at worst will trigger some undesired side effects (think bigger codebases)
- you will spend your whole career creating programs for others, which means that accurate error messages and clean program termination will be expected by your users.

Reply:
By the way, you need to close files whenever you're done with them.
You have two ways of doing so:
- either you call `f.Close()` whenever you're done with the file
- or you use `defer`, right after having opened it (recommended). https://gobyexample.com/defer